src/js/line.js

```diff
@@ -0,0 +1,521 @@
+                .attr({
+                    "cy": function (d) {
+                        return yScale(d.value);
+                    },
+                    "cx": function (d) {
+                        return xScale(new Date(d.date));
+                    }
+                });
+        });
+
+    };
+
+    floe.chartAuthoring.lineChart.chart.getYScale = function (that) {
+        var height = that.options.svgOptions.height;
+        var padding = that.options.lineOptions.padding;
+        var dataSet = that.model.wrappedDataSet;
+
+        var maxValues = [];
+
+        fluid.each(dataSet, function (setItem) {
+            var maxValue = d3.max(setItem.data, function (d) {
+                return d.value;
+            });
+            maxValues.push(maxValue);
+        });
+
+        var dataSetMax = d3.max(maxValues, function (d) {
+            return d;
+        });
+
+        return d3.scale.linear()
+            .domain([0, dataSetMax])
+            .nice()
+            .range([height - padding, padding]);
+    };
+
+    floe.chartAuthoring.lineChart.chart.getXScale = function (that) {
+        var width = that.options.svgOptions.width;
+        var padding = that.options.lineOptions.padding;
+        var dataSet = that.model.wrappedDataSet;
+
+        var minDates = [];
+        var maxDates = [];
+
+        fluid.each(dataSet, function (setItem) {
+            var minDate = d3.min(setItem.data, function (d) {
+                return new Date(d.date);
+            });
+            minDates.push(minDate);
+
+            var maxDate = d3.max(setItem.data, function (d) {
+                return new Date(d.date);
+            });
+            maxDates.push(maxDate);
+
+        });
+
+        var minDate = d3.min(minDates, function (d) {
+            return new Date(d);
+        });
+
+        var maxDate = d3.max(maxDates, function (d) {
+            return new Date(d);
+        });
+
+        return d3.time.scale()
+            .domain([minDate, maxDate])
+            .range([padding, width - padding * 2]);
+    };
+
+    floe.chartAuthoring.lineChart.chart.getYAxis = function (that) {
+        var width = that.options.svgOptions.width;
+        var padding = that.options.lineOptions.padding;
+        var yScale = floe.chartAuthoring.lineChart.chart.getYScale(that);
+
+        var yAxis = d3.svg.axis()
+            .orient("left")
+            .scale(yScale)
+            .innerTickSize(- width + padding * 3)
+            .outerTickSize(0)
+            .tickPadding(10);
+
+        return yAxis;
+    };
+
+    floe.chartAuthoring.lineChart.chart.getXAxis = function (that) {
+        var xScale = floe.chartAuthoring.lineChart.chart.getXScale(that),
+            numberOfXAxisTicks = that.options.lineOptions.numberOfXAxisTicks;
+
+        // See https://github.com/mbostock/d3/wiki/Time-Formatting for
+        // explanation of how time formatting works in D3
+        var customTickFormat = d3.time.format.multi([
```

Comment:
Why cut n paste this from the docs when we could just use "the default operated by d3.time.scale"? If we intend this to be truly configurable, we should break this out into some configuration accepted by the component, or at the very least an invoker which the user can configure (by itself, rather then being embedded in "getXAxis"